### PR TITLE
feat: Atkey add getKeyType

### PR DESCRIPTION
**Pull request by murali-shris**
**- What I did**
- added support in AtKey for validation method in persistence layer
**- How I did it**
- added a new method getKeyType which will return a valid key type enum from KeyType
- added a new enum value reservedKey. 
- added  new exception InvalidAtKeyException
- exported key_type.dart from at_commons
**- How to verify it**
- run the unit tests at_key_type_test.dart

### CODE REVIEW

**Comment by murali-shris** [comment]
Impl for ReservedKey  - Pending

File: at_commons/lib/src/keystore/at_key.dart [hunk 1]
```diff
@@ -210,6 +212,11 @@ class AtKey {
     atKey.metadata = metaData;
     return atKey;
   }
+
+  //# TODO dartdocs
```
Comment (VJag) [inline at_commons/lib/src/keystore/at_key.dart]: Dart docs

File: at_commons/test/at_key_type_test.dart [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import 'package:at_commons/at_commons.dart';
+import 'package:at_commons/src/keystore/key_type.dart';
+import 'package:test/test.dart';
+
+void main() {
```
Comment (VJag) [inline at_commons/test/at_key_type_test.dart:5]: Should the tests be extended for various reserved keys?

File: at_commons/lib/src/utils/at_key_regex_utils.dart [hunk 3]
```diff
@@ -22,11 +23,16 @@ class Regexes {
...
       '''(?<visibility>(cached:public:){1})((@(?<sharedWith>($_charsInAtSign|$_allowedEmoji){1,55}):))?(?<entity>($_charsInEntity|$_allowedEmoji)+)\\.(?<namespace>$_charsInNamespace)@(?<owner>($_charsInAtSign|$_allowedEmoji){1,55})''';
+  static const reservedKey =
+      '^($commitLogCompactionKey|$accessLogCompactionKey|$notificationCompactionKey|$AT_PKAM_PRIVATE_KEY|$AT_PKAM_PUBLIC_KEY|$AT_ENCRYPTION_SELF_KEY|$AT_CRAM_SECRET|$AT_CRAM_SECRET_DELETED|$AT_SIGNING_KEYPAIR_GENERATED)\$';
 }
 
 class RegexUtil {
   /// Returns a first matching key type after matching the key against regexes for each of the key type
   static KeyType keyType(String key) {
+    if (reservedKeys.contains(key)) {
+      return KeyType.reservedKey;
```
Comment (VJag) [inline at_commons/lib/src/utils/at_key_regex_utils.dart:115]: Is what specific reserved key be included?

**Comment by murali-shris** [comment]
> Impl for ReservedKey - Pending

done

File: at_commons/test/at_key_type_test.dart (hunk 2 above)
Comment (murali-shris) [reply at_commons/test/at_key_type_test.dart:5]: yes Jagan..I will take colin's prod hive box and run through all the keys and write tests to cover the missed scenarios

**Review by gkc — DISMISSED** [review]

File: at_commons/lib/src/keystore/at_key.dart (hunk 1 above)
Comment (murali-shris) [reply at_commons/lib/src/keystore/at_key.dart]: done

File: at_commons/lib/src/utils/at_key_regex_utils.dart (hunk 3 above)
Comment (gkc) [reply at_commons/lib/src/utils/at_key_regex_utils.dart:115]: I don't see a need for that right now?

**Review by gkc — APPROVED** [review]
LGTM ... I made a few commits (1) to make it easier to do the diff review (2) to make the code to do with reservedKey consistent with other key types (i.e. add as instance getter for Regexes, not a static)

**Pull request merged**